### ISSUE TITLE
Build error on Ionic Pro and VSTS: webpack-environment-suffix-plugin.js:78

### ISSUE THREAD

**Issue by blumk** [AksimO/webpack-environment-suffix-plugin#1]
Running your plugin on the Ionic Pro build server throws an error:

```
> ionic-app-scripts build

[08:51:00]  ionic-app-scripts 3.1.0 
[08:51:00]  build dev started ... 
[08:51:00]  clean started ... 
[08:51:00]  clean finished in 1 ms 
[08:51:00]  copy started ... 
[08:51:00]  deeplinks started ... 
[08:51:00]  deeplinks finished in 154 ms 
[08:51:00]  transpile started ... 
There was an error in config file "/usr/src/app/webpack.config.js". Using defaults instead.
/usr/src/app/node_modules/webpack-environment-suffix-plugin/src/webpack-environment-suffix-plugin.js:78
          newModule => /*  */ c(null, { ...r, request: newModule }),
                                        ^^^

SyntaxError: Unexpected token ...
    at createScript (vm.js:74:10)
    at Object.runInThisContext (vm.js:116:10)
    at Module._compile (module.js:533:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object. (/usr/src/app/webpack.config.js:3:33)
[08:51:05]  transpile finished in 4.41 s 
[08:51:05]  preprocess started ... 
[08:51:05]  preprocess finished in 1 ms 
[08:51:05]  webpack started ... 
There was an error in config file "/usr/src/app/webpack.config.js". Using defaults instead.
/usr/src/app/node_modules/webpack-environment-suffix-plugin/src/webpack-environment-suffix-plugin.js:78
          newModule => /*  */ c(null, { ...r, request: newModule }),
                                        ^^^

SyntaxError: Unexpected token ...
    at createScript (vm.js:74:10)
    at Object.runInThisContext (vm.js:116:10)
    at Module._compile (module.js:533:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object. (/usr/src/app/webpack.config.js:3:33)
[08:51:12]  copy finished in 12.15 s 
[08:51:12]  webpack finished in 7.55 s 
[08:51:12]  sass started ... 
[08:51:14]  sass finished in 1.23 s 
[08:51:14]  postprocess started ... 
[08:51:14]  postprocess finished in 9 ms 
[08:51:14]  lint started ... 
[08:51:14]  build dev finished in 13.49 s 
[08:51:17]  lint finished in 3.05 s
```

Same error happens when I try to build it on VSTS (build server running Windows):
```
2017-11-17T16:44:57.9298166Z [16:44:57]  ionic-app-scripts 3.1.0 
2017-11-17T16:44:57.9438161Z [16:44:57]  build dev started ... 
2017-11-17T16:44:57.9848170Z [16:44:57]  clean started ... 
2017-11-17T16:44:57.9868230Z [16:44:57]  clean finished in 2 ms 
2017-11-17T16:44:57.9868230Z [16:44:57]  copy started ... 
2017-11-17T16:44:58.1368160Z [16:44:58]  deeplinks started ... 
2017-11-17T16:44:58.4298261Z [16:44:58]  deeplinks finished in 274 ms 
2017-11-17T16:44:58.4328174Z [16:44:58]  transpile started ... 
2017-11-17T16:44:58.5618173Z There was an error in config file "d:\a\1\s\webpack.config.js". Using defaults instead.
2017-11-17T16:44:58.5618173Z d:\a\1\s\node_modules\webpack-environment-suffix-plugin\src\webpack-environment-suffix-plugin.js:78
2017-11-17T16:44:58.5618173Z           newModule => /*  */ c(null, { ...r, request: newModule }),
2017-11-17T16:44:58.5618173Z                                         ^^^
2017-11-17T16:44:58.5618173Z SyntaxError: Unexpected token ...
2017-11-17T16:44:58.5618173Z     at Object.exports.runInThisContext (vm.js:76:16)
2017-11-17T16:44:58.5618173Z     at Module._compile (module.js:542:28)
2017-11-17T16:44:58.5618173Z     at Object.Module._extensions..js (module.js:579:10)
2017-11-17T16:44:58.5618173Z     at Module.load (module.js:487:32)
2017-11-17T16:44:58.5618173Z     at tryModuleLoad (module.js:446:12)
2017-11-17T16:44:58.5628175Z     at Function.Module._load (module.js:438:3)
2017-11-17T16:44:58.5628175Z     at Module.require (module.js:497:17)
2017-11-17T16:44:58.5628175Z     at require (internal/module.js:20:19)
2017-11-17T16:44:58.5628175Z     at Object.<anonymous> (d:\a\1\s\webpack.config.js:3:33)
2017-11-17T16:44:58.5628175Z     at Module._compile (module.js:570:32)
2017-11-17T16:45:03.6508165Z [16:45:03]  transpile finished in 5.24 s 
2017-11-17T16:45:03.6508165Z [16:45:03]  preprocess started ... 
2017-11-17T16:45:03.6518194Z [16:45:03]  preprocess finished in 1 ms 
```

I'm using Ionic 3.9.2 and I'm using the webpack config described in "Ionic >= 3.8.0".

**Comment by AksimO** [comment]
@blumk, what node version are you using. It looks like node does not understand es6 [Spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/). It can be replaces with equivalent.
``` 
{ ...r, request: newModule }
``` 
is equal to 
```
Object.assign({}, r, { request: newModule });
```

**Comment by blumk** [comment]
I'm running v8.9.0 locally, both build agents use older versions:

Ionic Pro build agent:
```
Running: npm run build
npm info it worked if it ends with ok
npm info using npm@5.3.0
npm info using node@v8.2.1
```

VSTS build agent:
```
2017-11-21T15:29:34.7372248Z [command]C:\Windows\system32\cmd.exe /D /S /C ""C:\Program Files\nodejs\npm.cmd" config list"
2017-11-21T15:29:36.2682185Z ; cli configs
2017-11-21T15:29:36.2682185Z user-agent = "npm/3.10.8 node/v6.9.1 win32 x64"
```

Node should support the spread operator since v5.0: http://node.green/

**Comment by AksimO** [comment]
@blumk, I’m pretty sure it complains about spread operator in your case. I’m not sure why :( can you try to replace it with an alternative ```Object.assign({}, r, { request: newModule });```. I guess it will work or start to complain about arrow function.

**Comment by AksimO** [comment]
@blumk, I just realised that `{ ...r, request: newModule }` code is not part of `webpack.conf.js` but part of `webpack-environment-suffix-plugin`.  I have replaced it in a plugin. Could you try if it works for you with new `v0.1.2` version.

**Comment by blumk** [comment]
@AksimO The issue is fixed with release 0.1.2. Thank you!
